### DISCUSSION TITLE
Free roaming neutrals

### DISCUSSION THREAD

**Discussion by Danimal696** [flareteam/flare-game#866] [Q&A]
Is there a way to create a neutral that behaves like enemies but doesn´t attack you? Like roaming around and attacking hostiles, but doesn´t follow you around.
I know there is a conversion power, but that makes the creature your "slave".

**Comment by dorkster** [comment]
Sorry, I don't think there is a way to do this currently. We'd need to add support for allied creatures that don't follow the player.

**Reply by WithinAmnesia** [reply]
1.14?

**Reply by dorkster** [reply]
Possibly.

At this stage, 1.13 is taking a bit longer than anticipated. I may end up pushing some of the more drastic changes (i.e. the ones affecting player/enemy stats) to 1.14, instead focusing on getting 1.13 stable and out the door. If I can get 1.13 solid, I can better focus on new features in 1.14.

**Reply by WithinAmnesia** [reply]
1.13 December - January? 1.14 January - February?

I am hoping to get the new player stats systems working so I can work on itemization in my game for currently there just are not enough numbers without fractional values XD.

Although I would prefer quality > quantity although if these are going to be core systems moving forward the quality might be warranted. I am hoping to have the Flare Engine be able to support content 'spirit' / gameplay features (even if partiality) from classics such as Baldur's Gate, Everquest, WoW, Diable, Poe, D&D etc. challenging to do but doable. We are almost there in terms of new features in development. I think all I am missing for critical massive world building creative tools / feature support are everything on the 1.13 list and possibly these potential ~1.14 features:
1. https://github.com/flareteam/flare-game/discussions/867
2. #866 (This thread)
3. Flare-ize more Open Game Art world building art (Tilesets, Items, Creatures, more options for all graphics etc.). Although I can do more art content already. Although proprieties, critical core gameplay features working first then art and expansion; foundation first then structure second and then finally finishing and polishing.

Yet the real battle is here and now with getting the core of the gameplay features working. So as for myself personally to finally have access to the creative tools and features to implement grand scale medieval fantasy and game progression features that I have been trying to find for 3 decades. Basically Gameplay > Graphics.

From my past developing experiences (successes and failures alike) I have learned that it is extraordinarily hard to retrofit gameplay features and more over *Grand Scale* into an already built project. So knowing my past struggles with other game engines and the hell of retrofitting grand scale features to turn a tricycle into a helicopter. I would rather stop, get the ~'helicopter tools' to begin with and avoid the agony of having a project be mostly done, yet critically missing huge core elements. Then having to retrofit and literally smash features into old content (with my face against brick walls) effectively tripling (at least) the effort of content creation verses if the project was built with the grand scale features to begin with.

For being one person retrofitting feature hell is nigh-almost a guaranteed death sentence for any reasonable timeline of productivity for producing creative content XD. Of which I hope to learn from and work properly going forward. Thus I am waiting patiently before charging headlong into another series of brick walls XD. 

I really hope 1.13 and 1.14 can see the light of day and with all the work being on mostly your shoulders @dorkster I feel guilty asking so much. Yet at the same time I am kind of out of my element with coding currently (I feel like derrrrrr waayyy too much here XD). Also literally these 1.13 and 1.14 features are coalescing to become the last major critical step away from myself (at least) being finally free to create one hell of a kick ass R.P.G.; (if things stay stable and relatively sane in my life for the next 2-4 years *fingers crossed*).

With being one person trying to harness lots of creative power: Low input, high output is gold.

**Reply by dorkster** [reply]
I'm looking at early to mid January for 1.13. I'll be giving a short talk about Flare for FSFE's [I Love Free Software Day](https://fsfe.org/activities/ilovefs/index.en.html) this February, so I'd like to have 1.13 stable/released beforehand.

I can say with certainty that 1.14's development will be slower. I'll try to keep the feature list in check as to not end up with a 2-year dev period like 1.11 -> 1.12. Maybe that means some of the 1.14 stuff gets pushed to 1.15. At the end of the day, Flare is just a hobby for me. I have to look after my own well-being.

**Reply by WithinAmnesia** [reply]
I feel there is an issue related to the effort and well being of people like yourself @dorkster where without yourself this whole project kind of dies slowly. Yet without people like myself and others who use the tools to build video games with, the project ultimately may be argued to have no ehh ~'meaning' or wider outside / public use / something aspiring to be towards a ~'noble aim' in helping others?

So I have this loose unrefined thought of trying to improve the lives of people like yourself so that ultimately I do not end up as a parasite (or something close to said approximation) in this creative process / journey. I have past experiences that make myself hesitant to just throwing money out to try to fill that gaping void of guilt developing in the back of my mind; within this ongoing ambitious creative endeavour when addressing the sacrifice deficit and burden being placed on the shoulders of people like yourself battling year after year with serious coding issues. Demoralizingly, seemingly only being rewarded in thought and gesture or at worse platitudes of kindness while others refuse to return the generosity; like a parasitic toxic relationship.

I do not wish this significant possibility to occur to the end result of the people who's sacrifices enabled my possible ultimate success and potential creative outbreak onto the world scene / larger theater of the digital community at large. Thus I am currently at internal odds within my own psyche for on one hand I feel a hollow emptying void of a distance of reciprocal altruism / generosity that this whole system / creative endeavour is founded upon.

Thus I have my current emotional state as of where I situate my actions of potential reprisal / return of generosity to thank and reward such hard and critical work that is enabling of my own creative success be in question. Whereas it may increasingly be potentially currently thrown about by the pangs of murky waves of uncertainty. Also therein as by the anonymity that separates people such as myself and the wider online community masquerading behind a digital status quo; but one that may be without a grounded substance of face to face honesty.

As such I am torn to assume what you desire and people like yourself (such as Clint Bellanger) as the founders and facility providers of these great creative tools. Yet I feel the status quo is without complete justification of its existence in a sustainable manner moving forward towards more iterations and a further amassing wealth of work and realized creative vision. I too struggle in the present for my own well being and thus I pain to understand that potential struggles are going on outside of my virtual understanding and I wonder the story behind the story as to what endeavours you and the creative founders face; yet burden not the community in sharing?

I feel in my heart of hearts this growing gulf of generosity within the solitude of due reciprocity is potentially self destructive in the same latitudes of severities as the doom of parasite upon their hosts. Thus I feel that an effort should be taken upon myself to become forthright and shine a light in the darkness of this ever churning emotional struggle within myself in hopes that the transparency; and the light of outside understanding can disinfect this malignancy growing inside of my psyche at odds with itself. Yet in a hopefully constructive manner as to the aspiring hope to make a new system to better support the vital sacrifices and to better justify them for the critical creative founders of these creative endeavours for people such as yourself and Clint and the myriad of community who gave for all yet expecting in return nothing.

I feel an emptiness developing past a threshold of an emotional critical mass and I feel that I should address this concern rather than to hide from the fears of being consumed by a parasitic toxic relationship. That, if I am not careful herein, may expand to wrap its malignant tendrils around the creative community and consume the very beating heart of this creative endeavour. Thus I am at odds with my current inaction towards forward momentum towards of a sort of system of reciprocal reward for people like yourself.

Yet I am in a haze of uncertainty on how to proceed towards my aims to rewards people such as yourself. Yet I feel I must light a fire under my butt and refuse to be idle; I must improve this way of existence / forward momentum / creative progress. Yet I feel that I currently need to seek help from without in order to gain the necessary clarity to tackle such parasitic demons at the gates of creativity in order to win a battle for continued sustainable growth. I feel that I must be pressed onwards to this more noble or at least more fair status quo of due reciprocity towards the critical creative founders of this entire project.

I feel as though am deadlocked within an internal struggle to avoid another round of life's justified disapproval of the state a wider existence encircling myself; in as where my own veteranicy of such harrowing campaigns of strife can be beguilingly attested to by my own baring of deep physiological scars that speak only of the dreadfully costly survival past failures. Even for those who were not guilty; everyone paid the price for the failures of the few empowered by others' collective trust. Yet I hope to remind myself to not be idle herein but to preemptively strike first before more than just myself are potentially consumed by these devils in waiting: work demand abuse; burnout, ceaseless feature bloat. Death of a thousands cuts for the death of a thousand dreams.

**Reply by dorkster** [reply]
Don't worry too much about it. I'm just trying to keep expectations reasonable.

**Reply by WithinAmnesia** [reply]
Cool.

**Comment by Danimal696** [comment]
Then i guess, i could make a hostile with no attack "power" that runs away from you. Another solution would be a guardian tower with no movement but can shoot arrows, I was trying to make a guardian for a wilderness outpost that just killed nearby enemies and returned to his post.

**Reply by WithinAmnesia** [reply]
https://youtu.be/Z3GzbRFMl5A?t=290 Is this like the neutral barbarian NPC that attack demons in Diablo 2 where they are neutral and attack enemies but are not controlled by the player? Like a bunch of 'guards' that attack monsters nearby (an arbitrary radius / line of sight aggro radius / close to their spawn area) but are not controlled by the player? Possibly able for players to 'lure' monsters into them and have the ~'guards' kill the monsters for the player (but hopefully with no experience reward to the player to avoid experience exploits)?

**Reply by Danimal696** [reply]
Yep, that was my idea.
